akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcProjectionImpl.scala

```diff
@@ -588,23 +587,25 @@ private[projection] object R2dbcProjectionImpl {
 
     // This is similar to R2dbcProjectionImpl.replayIfPossible but difficult to extract common parts
     // since this is flow processing
-    def replayIfPossible(originalEnvelope: Envelope, observer: HandlerObserver[Envelope]): Future[ReplayResult] = {
+    def replayIfPossible(
```

Comment:
Did you see my comment in the issue?

> We must be careful to not emit addition Done for the replayed events to downstream, because there must be 1:1 of incoming envelope with outgoing Done. Can we correlate that?

Have you handled that already?

Reply:
This pretty much exactly like before, where the completion became a Future and mapAsync it is now a source->fold into a single element source that is flatMapConcat:ed into the main stream.

So I _think_ that is covered.

Reply:
Not pushing, because seems like useless extra overhead, but added a

```scala
val replayAtMostOneElementAssert = Flow[Option[(Envelope, ProjectionContext)]].statefulMap(() => 0)({
      (count, option) =>
        if (count > 1) throw new IllegalStateException("Replay flow returned more than one element")
        (count + 1, option)
    }, _ => None)
```

and ran the returned replay stream through that, test suite all green.

Reply:
We got the passing on of the original rejected envelope for correlation wrong originally. Fix was in #1330.

This looks correct, with the single element source in place of the future. And hopefully tests already cover the cases as well.